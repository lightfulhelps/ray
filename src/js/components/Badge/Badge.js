// @flow
import * as React from 'react';
import classNames from 'classnames';

type Props = {
  children: React.Element<any>,
  className?: string,
  color?: string,
  isPill?: boolean,
  tag?: string,
  theme?: string,
};

const Badge = ({
  children,
  className,
  color,
  isPill,
  tag: Tag = 'span',
  theme = 'primary',
}: Props) => {
  const classes = classNames(
    className,
    'badge',
    'text-uppercase',
    theme ? `badge-${theme}` : null,
    {
      'badge-pill': isPill,
    }
  );

  return (
    <Tag className={classes} style={{ backgroundColor: color }}>
      {children}
    </Tag>
  );
};

export default Badge;
