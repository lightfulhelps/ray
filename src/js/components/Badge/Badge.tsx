import * as React from 'react';
import classNames from 'classnames';

type Props = {
  children: React.ReactNode;
  className?: string;
  color?: string;
  isPill?: boolean;
  tag?: keyof JSX.IntrinsicElements | typeof React.Component;
  theme?: string;
  [key: string]: any;
};

const Badge: React.FC<Props> = ({
  children,
  className,
  color,
  isPill,
  tag: Tag = 'div',
  theme = 'primary',
  ...other
}: Props) => {
  const classes = classNames(className, 'badge', `bg-${theme}`, 'text-truncate', 'mw-100', {
    'rounded-pill': isPill,
  });

  return (
    <Tag {...other} className={classes} style={{ backgroundColor: color }}>
      {children}
    </Tag>
  );
};

export default Badge;
