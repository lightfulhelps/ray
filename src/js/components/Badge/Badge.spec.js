import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';
import Badge from './Badge';

const setup = (overrides = {}) => {
  const props = merge(
    {
      children: 'Badge',
    },
    overrides
  );
  const wrapper = shallow(<Badge {...props} />);

  return {
    wrapper,
    props,
  };
};

describe('<Badge />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle the children prop', () => {
    const { wrapper } = setup();

    expect(wrapper.children().text()).toEqual('Badge');
  });

  it('should handle the className prop', () => {
    const { wrapper } = setup();

    wrapper.setProps({ className: 'custom' });

    expect(wrapper.hasClass('badge')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle the color prop', () => {
    const { wrapper } = setup();

    wrapper.setProps({ color: '#ff0000' });

    expect(wrapper.prop('style')).toEqual({
      backgroundColor: '#ff0000',
    });
  });

  it('should handle the isPill prop', () => {
    const { wrapper } = setup();

    expect(wrapper.hasClass('badge-pill')).toBe(false);

    wrapper.setProps({ isPill: true });

    expect(wrapper.hasClass('badge-pill')).toBe(true);
  });

  it('should handle the tag prop', () => {
    const { wrapper } = setup();

    expect(wrapper.type()).toBe('span');

    wrapper.setProps({ tag: 'a' });

    expect(wrapper.type()).toBe('a');

    wrapper.setProps({ tag: 'div' });

    expect(wrapper.type()).toBe('div');
  });

  it('should handle the theme prop', () => {
    const { wrapper } = setup();

    expect(wrapper.hasClass('badge-primary')).toBe(true);

    wrapper.setProps({ theme: 'secondary' });

    expect(wrapper.hasClass('badge-primary')).toBe(false);
    expect(wrapper.hasClass('badge-secondary')).toBe(true);
  });
});
