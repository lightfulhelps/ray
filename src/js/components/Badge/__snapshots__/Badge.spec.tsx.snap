// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> should render 1`] = `
<div
  className="badge bg-primary text-truncate mw-100"
  style={
    Object {
      "backgroundColor": undefined,
    }
  }
>
  Badge
</div>
`;
